src/client/tile.cpp

```diff
@@ -1735,8 +1741,46 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			}
 
 			// Apply the "clean transparent" filter, if needed
-			if (m_mesh_texture_prefilter)
+			if (m_setting_mipmap)
```

Comment:
Only enabling `texture_clean_transparent` when mipmapping is enabled isn't enough. Bilinear or trilinear filtering alone also trigger the artifacts `texture_clean_transparent` is supposed to fix.

And since anisotropic filtering apparently forces linear filtering on some Intel GPUs (this was stated by erlehmann), it would make sense to also enable `texture_clean_transparent` when anistropic filtering is enabled.

The condition would be `mip_map || bilinear_filter || trilinear_filter || anisotropic_filter`. If this is too much of a change, you could do an actual revert instead and bring back the `texture_clean_transparent` setting.

Reply:
I don't disagree that we should fix it but this PR is really just meant to revert to the previous state.

Reply:
That makes sense, but for this to actually be a "just revert to the previous state" PR, I think you'd have to bring back `texture_clean_transparent` as well.

Reply:
> Did anyone actually enable such an option hidden in advanced settings in practice though?

I did.  I have had it enabled continuously since I introduced the setting, and have advised everyone to enable it any time they run into the "black borders" problem.

It was really important back when I used to play minetest_game, because a lot of textures there had been optimized with RGB info discarded in transparent pixels.  I haven't needed it as much for NodeCore because I can just control the image optimization process to ensure that it's never necessary.

Reply:
If we're sure enough that it won't accidentally cause more issues (I don't see how myself), I can add it as a separate commit to the PR. Then we'll include that in the release and everyone should be happy no matter which graphics settings they had.